### DISCUSSION TITLE
Poor fit of dynamical model on some genes

### DISCUSSION THREAD

**Discussion by ftucos** [theislab/scvelo#791] [Q&A] [answered]
For some genes the fit obtained from the dynamical model is poor (here you see the example of Gsta4). I can picture what should be the proper shape to fit but I don't get why the expectation–maximization algorithm settles on those parameters.
![image](https://user-images.githubusercontent.com/27356045/147704120-c05a7117-975e-4c68-a0e7-712819c3ddbc.png)

I tried both increasing `max_iter = 1000` and setting `fit_basal_transcription = True` but I got no improvements.
</details>

I do realize that the fit is tricky because the shape is collapsed to almost linear but do you have any suggestions to improve the fit rather than just revert back to stochastic modeling?

<details> <summary> Versions </summary>

```pytb
scvelo==0.2.5.dev56+g9ff8098  scanpy==1.8.2  anndata==0.7.8  loompy==3.0.6  numpy==1.20.3  scipy==1.7.2  matplotlib==3.5.0  sklearn==1.0.1  pandas==1.3.4  
```
</details>

**Comment by WeilerP** [comment, marked as answer]
@ftucos, as you already mentioned, fitting a model for this gene is not easy as it does not exhibit the characteristic almond shape. Without the coloring by cell type, you could even easily argue to fit only an up- or down-regulation, IMO. The problem is, I believe, the inferred switching point. AON, there is no way to improve this fit.

**Reply by ftucos** [reply]
Thank you for your reply.  I think that although not fitting the characteristic almond shape, I recognized it as a decent dynamical gene because of the looping shape (no cells in the center) but I do understand that it is not a parameter that the model takes into account in the fit. I will just discard genes like that ;-)
